File: Test/LibraryTest.fs
module LibraryTest

open NUnit.Framework
open GeomLibrary

[<TestCase([|1.0f; 2.0f; 2.0f; 1.0f|], [|2.0f; 2.0f; 1.0f; 1.0f|], Result = "This is a convex quadrangle")>]
[<TestCase([|-2.0f; 0.0f; 2.0f; 0.0f|], [|0.0f; 2.0f; 0.0f; -2.0f|], Result = "This is a convex quadrangle")>]
[<TestCase([|-3.0f; 1.0f; 0.0f; 1.0f|], [|1.0f; 1.0f; 0.0f; -1.0f|], Result = "This is not a convex quadrangle")>]
[<TestCase([|-1.0f; 0.0f; 1.0f; 1.0f|], [|2.0f; 0.0f; -2.0f; 1.0f|], Result = "This is a triangle")>]
[<TestCase([|-1.0f; 0.0f; 1.0f; 2.0f|], [|-1.0f; 0.0f; 1.0f; 2.0f|], Result = "This is a line")>]
[<TestCase([|1.0f; 1.0f; 1.0f; 1.0f|], [|2.0f; 2.0f; 2.0f; 2.0f|], Result = "This is a dot")>]
let test (x : float32 array, y : float32 array) =
    let lib = Library()
    lib.result(x, y)

File: Test/LibraryTest.cs
using GeomLibrary;
using NUnit.Framework;

namespace Test
{
    [TestFixture]
    public class LibraryTest
    {
        [TestCase(new float[4] { 1f, 1f, 1f, 1f }, new float[4] { 2f, 2f, 2f, 2f }, Result = "This is a dot")]
        [TestCase(new float[4] { 1f, 1f, 1f, 2f }, new float[4] { 1f, 1f, 1f, 0f }, Result = "This is a line")]
        [TestCase(new float[4] { -1f, 0f, 1f, 2f }, new float[4] { -1f, 0f, 1f, 2f }, Result = "This is a line")]
        [TestCase(new float[4] { 1f, 1f, 2f, 0f }, new float[4] { 2f, 2f, 0f, 0f }, Result = "This is a triangle")]
        [TestCase(new float[4] { -1f, 0f, 1f, 1f }, new float[4] { 2f, 0f, -2f, 1f }, Result = "This is a triangle")]
        [TestCase(new float[4] { 1f, 2f, 2f, 1f }, new float[4] { 2f, 2f, 1f, 1f }, Result = "This is a convex quadrangle")]
        [TestCase(new float[4] { -2f, 0f, 2f, 0f }, new float[4] { 0f, 2f, 0f, -2f }, Result = "This is a convex quadrangle")]
        [TestCase(new float[4] { -3f, -1f, 1f, 3f }, new float[4] { -1f, 2f, 2f, -1f }, Result = "This is a convex quadrangle")]
        [TestCase(new float[4] { -3f, 1f, 0f, 1f }, new float[4] { 1f, 1f, 0f, -1f }, Result = "This is not a convex quadrangle")]
        [TestCase(new float[4] { 1f, -1f, 2f, 1f }, new float[4] { -1f, 1f, 2f, 1f }, Result = "This is not a convex quadrangle")]
        public string test(float[] x, float[] y)
        {
            Library lib = new Library();
            return lib.result(x, y);
        }
    }
}
